jetty-client/src/main/java/org/eclipse/jetty/client/http/HttpSenderOverHTTP.java

```diff
@@ -191,17 +191,19 @@ public String toString()
         private final HttpExchange exchange;
         private final Callback callback;
         private final MetaData.Request metaData;
+        private final HttpChannelOverHTTP httpChannelOverHTTP;
         private ByteBuffer headerBuffer;
         private ByteBuffer chunkBuffer;
         private ByteBuffer contentBuffer;
         private boolean lastContent;
         private boolean generated;
 
-        public HeadersCallback(HttpExchange exchange, HttpContent content, Callback callback)
+        public HeadersCallback(HttpExchange exchange, HttpContent content, Callback callback, HttpChannelOverHTTP httpChannelOverHTTP)
         {
             super(false);
             this.exchange = exchange;
             this.callback = callback;
+            this.httpChannelOverHTTP = httpChannelOverHTTP;
```

Comment:
You only need the HttpConnection, so keep that as the field and avoid two lookups on every write